# Maternal Health

In [1]:
# import libraries
import pandas as pd
import numpy as np

In [3]:
df = pd.read_csv(r'E:\Projects\Maternal-Mortality-Risk-Assessment-and-Intervention-Project\Data\Maternal Health Risk Data Set.csv')
df.head()

Age  SystolicBP  DiastolicBP    BS  BodyTemp  HeartRate  RiskLevel
0   25         130           80  15.0      98.0         86  high risk
1   35         140           90  13.0      98.0         70  high risk
2   29          90           70   8.0     100.0         80  high risk
3   30         140           85   7.0      98.0         70  high risk
4   35         120           60   6.1      98.0         76   low risk

### Data Exploration

In [5]:
class DataExplorer:
    def __init__(self, data):
        self.data = data
        self.shape = data.shape
        self.data_types = data.dtypes
        self.missing_values = data.isnull().sum()
        
    def summary(self):
        print("Dataset Shape:")
        print(self.shape)
        print("\nData Types:")
        print(self.data_types)
        print("\nMissing Values:")
        print(self.missing_values)

    def summary_statistics(self):
        numeric_data = self.data.select_dtypes(include='number')
        statistics = numeric_data.describe()
        print("\nSummary Statistics for Numerical Columns:")
        print(statistics)
        
    def unique_values(self):
        categorical_data = self.data.select_dtypes(exclude='number')
        unique_values = {}
        for column in categorical_data.columns:
            unique_values[column] = {
                'Unique Values': categorical_data[column].unique(),
                'Value Counts': categorical_data[column].value_counts()
            }
        print("\nUnique Values and Value Counts for Categorical Columns:")
        for column, values in unique_values.items():
            print(f"Column: {column}")
            print(f"Unique Values: {values['Unique Values']}")
            print(f"Value Counts:\n{values['Value Counts']}\n")


# Creating an instance of the DataExplorer class and explore the dataset
data_explorer = DataExplorer(df)
data_explorer.summary()
data_explorer.summary_statistics()
data_explorer.unique_values()

Dataset Shape:
(1014, 7)

Data Types:
Age              int64
SystolicBP       int64
DiastolicBP      int64
BS             float64
BodyTemp       float64
HeartRate        int64
RiskLevel       object
dtype: object

Missing Values:
Age            0
SystolicBP     0
DiastolicBP    0
BS             0
BodyTemp       0
HeartRate      0
RiskLevel      0
dtype: int64

Summary Statistics for Numerical Columns:
               Age   SystolicBP  DiastolicBP           BS     BodyTemp  \
count  1014.000000  1014.000000  1014.000000  1014.000000  1014.000000   
mean     29.871795   113.198225    76.460552     8.725986    98.665089   
std      13.474386    18.403913    13.885796     3.293532     1.371384   
min      10.000000    70.000000    49.000000     6.000000    98.000000   
25%      19.000000   100.000000    65.000000     6.900000    98.000000   
50%      26.000000   120.000000    80.000000     7.500000    98.000000   
75%      39.000000   120.000000    90.000000     8.000000    98.000000   
max

- The dataset consists of 1,014 rows and 7 columns.

- The columns have the following data types:
  - Age, SystolicBP, DiastolicBP, and HeartRate are of integer type.
  - BS (Blood Sugar) and BodyTemp (Body Temperature) are of float type.
  - RiskLevel is of object type, representing categorical data.
  

- There are no missing values in any of the columns; all columns have complete data.

- The summary statistics provide valuable insights into the key numerical features of our dataset. The mean age of pregnant women in the dataset is approximately 29.87 years, with a standard deviation of 13.47, indicating a relatively wide age range. Systolic blood pressure (SystolicBP) has an average value of 113.20 mmHg, while diastolic blood pressure (DiastolicBP) averages around 76.46 mmHg. These blood pressure measures exhibit variation, as seen in their respective standard deviations. Blood sugar levels (BS) have an average molar concentration of 8.73 mmol/L, with a moderate standard deviation of 3.29, suggesting variability in glucose levels. Body temperature (BodyTemp) demonstrates a consistent value of 98.67°F, and the heart rate (HeartRate) has an average of 74.30 bpm, indicating a typical resting heart rate.
